src/main/java/mate/academy/daobook/DaoBookImpl.java

```
@@ -0,0 +1,154 @@
+import mate.academy.lib.Dao;
+import mate.academy.model.Book;
+
+@Dao
+public class DaoBookImpl implements DaoBook {
+
+    public Book update(Book book) {
+        String sql = "UPDATE books SET title = ?, price = ? WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(sql)) {
+            preparedStatement.setString(1, book.getTitle());
+            preparedStatement.setBigDecimal(2, book.getPrice());
+            preparedStatement.setLong(3, book.getId());
+            int affectedRows = preparedStatement.executeUpdate();
+            if (affectedRows < 1) {
+                throw new RuntimeException(
+                "Expected to insert at least one row, but inserted 0 rows");
+            }
+            Optional<Book> updatedBook = findById(book.getId());
+            if (updatedBook.isPresent()) {
+                return updatedBook.get();
+            }
+        } catch (SQLException e) {
+            throw new RuntimeException(e);
+        } catch (DataProcessingException ex) {
+            throw new RuntimeException("Can't update book" + book, ex);
+        }
+        return null;
+    }
+
+    public boolean deleteById(Long id) {
+        String sql = "DELETE FROM books WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(sql)) {
+            preparedStatement.setLong(1, id); {
+                int affectedRows = preparedStatement.executeUpdate();
+                if (affectedRows < 1) {
+                    throw new RuntimeException(
+                            "Expected to insert at least one row, but inserted 0 rows");
+                }
+                return true;
+            }
+        } catch (DataProcessingException e) {
+            throw new RuntimeException("Can't delete book by id with id " + id, e);

```

You catch DataProcessingException in some places and wrap it into RuntimeException (lines 37-38, 55-56). The checklist/description require DAO layer to rethrow DataProcessingException with informative messages instead of converting it to plain RuntimeException. Quote: "You should rethrow this exception in `catch` block on dao layer." Please change these catch blocks to throw DataProcessingException (or wrap SQLException into DataProcessingException with contextual text).